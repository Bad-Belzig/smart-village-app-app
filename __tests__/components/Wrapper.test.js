jest.setTimeout(30000);
import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';

import {
  Wrapper,
  WrapperHorizontal,
  WrapperVertical,
  WrapperLandscape,
  WrapperRow,
  WrapperWrap,
  InfoBox,
  WrapperWithOrientation
} from '../../src/components';

describe('testing Wrapper style component', () => {
  it('renders a default Wrapper', async () => {
    const tree = renderer.create(<Wrapper />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperHorizontal', () => {
    const tree = renderer.create(<WrapperHorizontal big />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperVertical', async () => {
    const tree = renderer.create(<WrapperVertical />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperLandscape', async () => {
    const tree = renderer.create(<WrapperLandscape noFlex />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperRow', async () => {
    const tree = renderer.create(<WrapperRow />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperRow that centers items', async () => {
    const tree = renderer.create(<WrapperRow center />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperRow that does spaceAround items', async () => {
    const tree = renderer.create(<WrapperRow spaceAround />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperRow that does spaceBetween items', async () => {
    const tree = renderer.create(<WrapperRow spaceBetween />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperWrap', async () => {
    const tree = renderer.create(<WrapperWrap />);
    expect(tree).toMatchSnapshot();
  });

  it('renders InfoBox', async () => {
    const tree = renderer.create(<InfoBox />);
    expect(tree).toMatchSnapshot();
  });

  it('renders WrapperWithOrientation', async () => {
    const tree = renderer.create(
      <WrapperWithOrientation needLandscapeWrapper>
        <View></View>
      </WrapperWithOrientation>
    );
    expect(tree).toMatchSnapshot();
  });
});
