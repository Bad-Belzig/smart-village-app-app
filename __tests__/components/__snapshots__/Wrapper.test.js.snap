// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Wrapper style component renders InfoBox 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginBottom": 5.6000000000000005,
      },
    ]
  }
/>
`;

exports[`testing Wrapper style component renders WrapperHorizontal 1`] = `
<View
  big={true}
  style={
    Array [
      Object {
        "paddingLeft": 19.040000000000003,
        "paddingRight": 19.040000000000003,
      },
    ]
  }
/>
`;

exports[`testing Wrapper style component renders WrapperLandscape 1`] = `
<View
  noFlex={true}
  style={
    Array [
      Object {
        "paddingLeft": "15%",
        "paddingRight": "15%",
      },
    ]
  }
/>
`;

exports[`testing Wrapper style component renders WrapperRow 1`] = `
<View
  center={true}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`testing Wrapper style component renders WrapperRow 2`] = `
<View
  spaceAround={true}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      },
    ]
  }
/>
`;

exports[`testing Wrapper style component renders WrapperVertical 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 15.680000000000001,
        "paddingTop": 15.680000000000001,
      },
    ]
  }
/>
`;

exports[`testing Wrapper style component renders WrapperWithOrientation 1`] = `
<View
  noFlex={false}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingLeft": "15%",
        "paddingRight": "15%",
      },
    ]
  }
>
  <View />
</View>
`;

exports[`testing Wrapper style component renders WrapperWrap 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "width": "100%",
      },
    ]
  }
/>
`;

exports[`testing Wrapper style component renders a default Wrapper 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 15.680000000000001,
        "paddingLeft": 15.680000000000001,
        "paddingRight": 15.680000000000001,
        "paddingTop": 15.680000000000001,
      },
    ]
  }
/>
`;
